packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/oidcprovider.ts

```diff
@@ -0,0 +1,149 @@
+import { ResourceHandler } from "./common";
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as aws from "aws-sdk";
+import * as https from "https";
+
+export class OpenIDConnectProviderResourceHandler extends ResourceHandler {
+  protected async onCreate() {
+    const clusterName = this.event.ResourceProperties.Config.clusterName;
+
+    const cluster = await this.eks.describeCluster({ name: clusterName });
+    if (!cluster) {
+      throw new Error(`no cluster found with name ${clusterName}`);
+    }
+
+    const issuerUrl = cluster.cluster?.identity?.oidc?.issuer;
+    if (!issuerUrl) {
+      throw new Error(`no issuer found on cluster`);
+    }
+
+    const thumbprint = await this.getIssuerCAThumbprint(issuerUrl);
+    const createOpenIDConnectProviderResponse = await this.eks.createOpenIDConnectProvider(
+      {
+        Url: issuerUrl,
+        ClientIDList: ["sts.amazonaws.com"],
+        ThumbprintList: [thumbprint]
+      }
+    );
+
+    return {
+      PhysicalResourceId: createOpenIDConnectProviderResponse.OpenIDConnectProviderArn,
+      Data: {
+        openIDConnectIssuerUrl: issuerUrl.substring(8), // Strip https:// from the issuer
+        openIDConnectProviderArn: createOpenIDConnectProviderResponse.OpenIDConnectProviderArn
+      }
+    };
+  }
+
+  protected async onDelete() {
+    if (!this.physicalResourceId) {
+      throw new Error(
+        `Cannot delete a profile without a physical id`
+      );
+    }
+
+    const deleteOpenIDConnectProvider: aws.IAM.DeleteOpenIDConnectProviderRequest = {
+      OpenIDConnectProviderArn: this.physicalResourceId
+    };
+
```

Comment:
Maybe call "checkResource" first and return success if it's not found (idempotency)